src/imageTools/highlight.js

```diff
@@ -83,9 +88,15 @@ var cornerstoneTools = (function ($, cornerstone, cornerstoneMath, cornerstoneTo
 
         context.save();
         var data = toolData.data[0];
-
-        var selectionColor="white",
-            toolsColor="white";
+        if (!data) return;
+
+        if (pointNearTool(data, cornerstoneTools.toolCoordinates.getCoords())) {
```

Comment:
You can use the pointInsideRect function to check if the cursor is inside the highlight tool.

Something like this should work:

````javascript
var coords = cornerstoneTools.toolCoordinates.getCoords();
if (pointNearTool(data, coords) || pointInsideRect(data, coords))
{
...
}...
````